### PR TITLE
[GH-15865] Upgrade org.python:jython to CWE-416 of com.github.jnr:jnr-posix (overriding org.python.core.imp class  from Jython with custom changes, search for CUSTOM CHANGE)

### LINKED ISSUE #15865: Upgrade org.python:jython to CWE-416 of com.github.jnr:jnr-posix
Support ticket: https://support.h2o.ai/a/tickets/106681

Reported vulnerability:
```
— Use After Free [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGITHUBJNR-1570422] in com.github.jnr:jnr-posix@3.0.23
    introduced by unknown:/opt/.venv/lib/python3.11/site-packages/h2o/backend/bin/h2o.jar@unknown > com.github.jnr:jnr-posix@3.0.23
```

It looks that `com.github.jnr:jnr-posix` dependency is brought to h2o.jar via `org.python:jython`. Although gradle doesn't see the dependency:
```
./gradlew :h2o-assemblies:main:dependencyInsight --dependency jnr-posix --configuration runtimeClasspath

> Task :h2o-assemblies:main:dependencyInsight
No dependencies matching given input were found in configuration ':h2o-assemblies:main:runtimeClasspath'
```

jnr-posix seems to be jython dependency. From `META-INF/maven/com.github.jnr/jnr-posix/pom.properties` inside Jython jar:
```
#Generated by Maven
#Thu Nov 12 15:35:48 CST 2015
version=3.0.23
groupId=com.github.jnr
artifactId=jnr-posix
```

### PR DESCRIPTION
**Pull request by mn-mikke**
Closes #15865 

See #15865 for the details about problem. `jnr-posix` is not reported by gradle as other transitive dependencies, but with the upgrade of Jython to 2.7.3, we get the following from ```META-INF/maven/com.github.jnr/jnr-posix/pom.properties``` of h2o.jar
```
#Generated by Maven
#Thu Jan 06 10:51:33 CST 2022
groupId=com.github.jnr
artifactId=jnr-posix
version=3.1.15
```

The version jnr-posix:3.1.15 shouldn't suffer from the vulnerability. (https://security.snyk.io/vuln/SNYK-JAVA-COMGITHUBJNR-1570422)

### CODE REVIEW

**Review by krasinski — APPROVED** [review]
:+1:

**Review by valenad1 — DISMISSED** [review]
Good finding, thanks!

File: h2o-extensions/jython-cfunc/src/main/java/org/python/core/imp.java [hunk 1]
```diff
@@ -0,0 +1,1791 @@
...
+ * Utility functions for "import" support.
+ *
+ * Note that this class tries to match the names of the corresponding functions from CPython's
+ * Python/import.c. In these cases we use CPython's function naming style (underscores and all
+ * lowercase) instead of Java's typical camelCase style so that it's easier to compare with
+ * import.c.
+ */
+public class imp {
+
+    private static Logger logger = Logger.getLogger("org.python.import");
```
Comment (valenad1) [inline h2o-extensions/jython-cfunc/src/main/java/org/python/core/imp.java:32]: `private static final Logger LOG = `

I would prefere.

Comment (valenad1) [reply h2o-extensions/jython-cfunc/src/main/java/org/python/core/imp.java:32]: Copy pased from original.. nothing to change

**Review by valenad1 — APPROVED** [review]
Thank you!

**Pull request merged**